src/BlockEntities/DispenserEntity.cpp

```diff
@@ -39,177 +41,244 @@ void cDispenserEntity::DropSpenseFromSlot(cChunk & a_Chunk, int a_SlotNum)
 	int BlockZ = (DispZ + DispChunk->GetPosZ() * cChunkDef::Width);
 
 	// Dispense the item:
-	switch (m_Contents.GetSlot(a_SlotNum).m_ItemType)
+	const cItem & SlotItem = m_Contents.GetSlot(a_SlotNum);
+	if (ItemCategory::IsMinecart(SlotItem.m_ItemType) && IsBlockRail(DispBlock))  // only actually place the minecart if there are rails!
 	{
-		case E_ITEM_BUCKET:
+		if (m_World->SpawnMinecart(BlockX + 0.5, DispY + 0.5, BlockZ + 0.5, SlotItem.m_ItemType) != cEntity::INVALID_ID)
 		{
-			LOGD("Dispensing empty bucket in slot %d; DispBlock is \"%s\" (%d).", a_SlotNum, ItemTypeToString(DispBlock).c_str(), DispBlock);
-			switch (DispBlock)
+			m_Contents.ChangeSlotCount(a_SlotNum, -1);
+		}
+	}
```

Comment:
There's no code following this, so how about bailing out with a `return`, so that the nesting level for the rest of the code is lower?